add parse capabilities for duration on event command

mdaffad opened the pull request:
## Description
<!-- Provide a clear and concise description of your changes -->

## Type of Change
<!-- Mark the appropriate option with an [x] -->
- [ ] Bug fix (non-breaking change that fixes an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] Performance improvement
- [ ] Test updates
- [ ] CI/CD related changes
- [ ] Other (please describe):

## Testing
<!-- Describe the tests you ran and/or added to verify your changes -->
- [ ] Added new unit tests
- [ ] Added new integration tests
- [ ] Ran existing test suite
- [ ] Tested manually (please provide steps)

## Related Issues
<!-- Link any related issues using the GitHub issue syntax: #issue-number
If there is no open issue for this, please add details of the problem or open a new issue. -->
Close #3206

## Additional Context
<!-- Add any other context about the pull request here -->

mdaffad committed:
add parse capabilities for duration on event command

Signed-off-by: Muhammad Daffa Dinaya <muhammaddaffadinaya@gmail.com>

mdaffad committed:
fix container event test error

Signed-off-by: Muhammad Daffa Dinaya <muhammaddaffadinaya@gmail.com>

mdaffad committed:
fix import Duration

Signed-off-by: Muhammad Daffa Dinaya <muhammaddaffadinaya@gmail.com>

mdaffad committed:
fix reorder error on import

Signed-off-by: Muhammad Daffa Dinaya <muhammaddaffadinaya@gmail.com>

YJDoc2 commented:
Hey, can you rebase and fix the clippy issues? Thanks :)

utam0k commented:
@mdaffad Please sign following the instructions in the link
https://github.com/youki-dev/youki/pull/3214/checks?check_run_id=47689260136

mdaffad committed:
fix: lint on clippy issues

Signed-off-by: Muhammad Daffa Dinaya <muhammaddaffadinaya@gmail.com>

mdaffad committed:
feat: utilize parse_duration

Signed-off-by: Muhammad Daffa Dinaya <muhammaddaffadinaya@gmail.com>

mdaffad committed:
chore: fix lint

Signed-off-by: Muhammad Daffa Dinaya <muhammaddaffadinaya@gmail.com>

utam0k commented:
@mdaffad May I ask you to address DCO?
https://github.com/youki-dev/youki/pull/3214/checks?check_run_id=48316005413

utam0k commented on crates/liboci-cli/Cargo.toml:
```diff
@@ -10,6 +10,9 @@ authors = ["youki team"]
 keywords = ["youki", "container", "oci"]
 
+[dependencies]
+parse_duration = "2.1.1"
```
Can't we implement it ourselves?

mdaffad committed:
drop parse_duration

Signed-off-by: Muhammad Daffa Dinaya <muhammaddaffadinaya@gmail.com>

mdaffad committed:
chore: fix lint

Signed-off-by: Muhammad Daffa Dinaya <muhammaddaffadinaya@gmail.com>

nayuta723 commented on crates/liboci-cli/src/events.rs:
```diff
@@ -1,11 +1,82 @@
+    Ok(Duration::from_secs(total_seconds))
+}
+
+fn parse_interval(s: &str) -> Result<Duration, String> {
```
Please update the return value to use an `Error` type instead of a `String`, to stay consistent with the implementation in `exec.rs`.
```suggestion
fn parse_interval(s: &str) -> Result<Duration, Box<dyn Error + Send + Sync + 'static>> {
```

ref: https://github.com/youki-dev/youki/blob/main/crates/liboci-cli/src/exec.rs#L67-L92

nayuta723 reviewed (commented):
@mdaffad 
Thank you for your contribution! 
To confirm that this change resolves Issue #3206, could you please post the test results in this PR?

mdaffad committed:
events: reject unit-less interval and add duration unit tests

Address review feedback on the events --interval parser:

- Drop the digit-only fast path so a unit-less value such as `1` is
  rejected, matching runc which relies on Go's time.ParseDuration.
- Return Box<dyn Error + Send + Sync + 'static> instead of String, for
  consistency with exec.rs.
- Parse with string slices (split_at / remainder slice) instead of a
  per-character String buffer, avoiding repeated allocations.
- Support sub-second units (ms, us/µs, ns) so values like `100ms` work.
- Add unit tests covering single, compound and invalid durations.

Signed-off-by: Muhammad Daffa Dinaya <muhammaddaffadinaya@gmail.com>

mdaffad committed:
Merge remote-tracking branch 'upstream/main' into enhancing-runc-compatibility-for-interval-on-event

Signed-off-by: Muhammad Daffa Dinaya <muhammaddaffadinaya@gmail.com>

# Conflicts:
#	crates/liboci-cli/src/events.rs

nayuta723 commented on crates/libcontainer/src/container/container_events.rs:
```diff
@@ -23,11 +24,11 @@ impl Container {
     /// .build()?;
     ///
-    /// container.events(5000, false)?;
+    /// container.events(Duration::from_secs(5000), false)?;
```
```suggestion
    /// container.events(Duration::from_secs(5), false)?;
```

nayuta723 commented on crates/liboci-cli/src/events.rs:
```diff
@@ -1,15 +1,137 @@
+    ("h", 3_600_000_000_000),
+    ("m", 60_000_000_000),
+    ("s", 1_000_000_000),
+];
```
The following order seems better to me. Are there any potential issues with it?

```suggestion
const UNITS: &[(&str, u128)] = &[
    ("ns", 1),
    ("us", 1_000),
    ("µs", 1_000),
    ("ms", 1_000_000),
    ("s", 1_000_000_000),
    ("m", 60_000_000_000),
    ("h", 3_600_000_000_000),
];
```
